sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/ProjectOverviewView.groovy

```diff
@@ -65,4 +86,19 @@ class ProjectOverviewView extends VerticalLayout{
         def dataProvider = new ListDataProvider(viewModel.projectOverviews)
         projectGrid.setDataProvider(dataProvider)
     }
+
+    private DownloadProjectComponent getDownloadComponent(ProjectSummary projectSummary) {
+        // see if the component was generated
+        if (downloadComponents.containsKey(projectSummary)) {
+            // we can reuse the component
+            return downloadComponents.get(projectSummary)
+        } else {
+            // remove all old information on the project summary form the map
+            downloadComponents.removeAll { (it.getKey().getCode() == projectSummary.getCode()) }
+            // generate new component
+            DownloadProjectComponent generatedComponent = DownloadProjectComponent.from(projectSummary, projectDownload)
+            downloadComponents.put(projectSummary, generatedComponent)
+            return generatedComponent
+        }
+    }
```

Comment:
If i understood it correctly, this method basically avoids that the button component is generated multiple times, which is an awesome idea! 🧠 
I'm curious to know, if you noticed a difference in loading time before and after this method was implemented.

Reply:
I did not notice a difference in loading time. The thing is, that the component renderer recreates these components on every selection, scroll, ...

The vaadin documentation https://vaadin.com/docs/v8/framework/components/components-grid/#components.grid.renderer specifically mentions this and recommends doing it this way. This is why I implemented it.